src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.ExpressionResult.cs

```diff
@@ -130,19 +130,31 @@ private static (ITypeSymbol? typeSymbol, bool returnsByRef) GetRegularExpression
                     return !info.Type.IsObjectType() ? info.GetTypeWithAnnotatedNullability() : info.GetConvertedTypeWithAnnotatedNullability();
                 }
             }
-        }
 
-        private static bool IsCoClassImplicitConversion(TypeInfo info, Conversion conversion, INamedTypeSymbol? coclassSymbol)
-        {
-            if (!conversion.IsImplicit ||
-                 info.ConvertedType == null ||
-                 info.ConvertedType.TypeKind != TypeKind.Interface)
+            private static bool IsCoClassImplicitConversion(TypeInfo info, Conversion conversion, INamedTypeSymbol? coclassSymbol)
             {
-                return false;
+                if (!conversion.IsImplicit ||
+                     info.ConvertedType == null ||
+                     info.ConvertedType.TypeKind != TypeKind.Interface)
+                {
+                    return false;
+                }
+
+                // let's see whether this interface has coclass attribute
+                return info.ConvertedType.HasAttribute(coclassSymbol);
             }
 
-            // let's see whether this interface has coclass attribute
-            return info.ConvertedType.HasAttribute(coclassSymbol);
+            public override SyntaxNode GetOutermostCallSiteContainerToProcess(CancellationToken cancellationToken)
```

Comment:
this wasa  method in the base class that specialized itself for the expr vs statement case.  but we have thse subclasses for that purpose.  so i broke the method  up and moved each part to the right subclass.